src/Pos/Worker.hs

```diff
@@ -35,7 +35,10 @@ allWorkers
 allWorkers = mconcatPair
     [
       -- Only workers of "onNewSlot" type
+
+      -- TODO cannot have this DHT worker here. It assumes Kademlia.
```

Comment:
Yes `WorkMode` assumes `MonadDHT` and `MonadDHT` is basically fixed to Kademlia. I'm working on abstracting the discovery mechanism. It's important that we can swap Kademlia out in favour of, for instance, a fixed set of peers. Helps with testing and profiling. So my idea here is to remove these listeners from `allListeners` and instead have them included by some `Pos.Launcher` module which chooses Kademlia.